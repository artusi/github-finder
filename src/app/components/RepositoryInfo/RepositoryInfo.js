import React, { Component } from "react";
import { connect } from "react-redux";
import { bindActionCreators } from "redux";
import PropTypes from "prop-types";
import { requestRepository } from "store/state/github";
import styles from "./RepositoryInfo.module.css";

export class RepositoryInfo extends Component {
  componentDidMount() {
    const { username, repositoryName } = this.props;

    this.props.requestRepository(username, repositoryName);
  }

  render() {
    const { username, repositoryName } = this.props;
    return (
      <div className={styles.repoInfo}>
        Hey you are seeing
        <span className={styles.infoItem}>{`  ${repositoryName} `}</span>
        from
        <span className={styles.infoItem}>{` ${username}`}</span>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  github: state.github
});

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      requestRepository
    },
    dispatch
  );
}

RepositoryInfo.propTypes = {
  username: PropTypes.string.isRequired,
  repositoryName: PropTypes.string.isRequired,
  requestRepository: PropTypes.func.isRequired
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(RepositoryInfo);
