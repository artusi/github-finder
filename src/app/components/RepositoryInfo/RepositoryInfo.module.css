.repoInfo {
  padding: 16px;
  border-radius: 12px;
  background: #f37a6f;
  max-width: 420px;
}
.infoItem {
  color: #00324d;
  font-weight: bold;
}

@media (max-width: 400px) {
  .repoInfo {
    width: 220px;
  }
}
