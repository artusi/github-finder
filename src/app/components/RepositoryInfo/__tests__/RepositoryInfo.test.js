import React from "react";
import { RepositoryInfo } from "../RepositoryInfo";

describe("<RepositoryInfo />", () => {
  it("renders", () => {
    const wrapper = shallow(
      <RepositoryInfo
        requestRepository={() => {}}
        username="artusi"
        repositoryName="another-repo"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("renders > with initial user", () => {
    const wrapper = shallow(
      <RepositoryInfo
        requestRepository={() => {}}
        username="artusi"
        repositoryName="another-repo"
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
