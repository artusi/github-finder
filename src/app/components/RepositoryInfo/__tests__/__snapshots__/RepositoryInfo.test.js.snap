// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryInfo /> renders > with initial user 1`] = `
<div
  className="repoInfo"
>
  Hey you are seeing
  <span
    className="infoItem"
  >
      another-repo 
  </span>
  from
  <span
    className="infoItem"
  >
     artusi
  </span>
</div>
`;

exports[`<RepositoryInfo /> renders 1`] = `
<div
  className="repoInfo"
>
  Hey you are seeing
  <span
    className="infoItem"
  >
      another-repo 
  </span>
  from
  <span
    className="infoItem"
  >
     artusi
  </span>
</div>
`;
